run.sh

```diff
@@ -15,17 +15,28 @@ while [ -n "$1" ]; do
   shift
 done
 
-SDC_HOME=${SDC_HOME:-$(mktemp -d)}
-
+if [ -d "$SDC_HOME" ]; then
+  SDC_HOME=${SDC_HOME}
+else
+  SDC_HOME="/tmp/sdc-home"
+  mkdir -p "SDC_HOME"
+  chmod 0700 "$SDC_HOME"
+fi
 export SDC_HOME
 
 GPG_HOME="$SDC_HOME/gpg"
 mkdir -p "$GPG_HOME"
-chmod 0700 "$SDC_HOME" "$GPG_HOME"
+chmod 0700 "$GPG_HOME"
 
 echo "Running app with home directory: $SDC_HOME"
 echo ""
 
+# make sure a ew gpg-agent is used for each run
+PID=$(ps -ef | grep gpg-agent | grep "$GPG_HOME" | grep -v grep | awk '{print $2}')
```

Comment:
nice! 😎 

what about using this in a bash trap (so that we can use `mktemp` above), i.e. here's a diff doing that based on your commit here:

```diff
diff --git a/run.sh b/run.sh
index 28fc9e3..e72b344 100755
--- a/run.sh
+++ b/run.sh
@@ -15,28 +15,25 @@ while [ -n "$1" ]; do
   shift
 done

-if [ -d "$SDC_HOME" ]; then
-  SDC_HOME=${SDC_HOME}
-else
-  SDC_HOME="/tmp/sdc-home"
-  mkdir -p "SDC_HOME"
-  chmod 0700 "$SDC_HOME"
-fi
+SDC_HOME=${SDC_HOME:-$(mktemp -d)}
 export SDC_HOME

 GPG_HOME="$SDC_HOME/gpg"
-mkdir -p "$GPG_HOME"
+mkdir -p "$SDC_HOME" "$GPG_HOME"
 chmod 0700 "$GPG_HOME"

+function cleanup {
+  # make sure a new gpg-agent is used for each run
+  PID=$(ps -ef | grep gpg-agent | grep "$GPG_HOME" | grep -v grep | awk '{print $2}')
+  if [ "$PID" ]; then
+    kill $PID
+  fi
+}
+trap cleanup EXIT
+
 echo "Running app with home directory: $SDC_HOME"
 echo ""

-# make sure a ew gpg-agent is used for each run
-PID=$(ps -ef | grep gpg-agent | grep "$GPG_HOME" | grep -v grep | awk '{print $2}')
-if [ "$PID" ]; then
-  kill "$PID"
-fi
-
 gpg --homedir "$GPG_HOME" --allow-secret-key-import --import tests/files/securedrop.gpg.asc &

 # create the database and config for local testing
@@ -44,4 +41,6 @@ gpg --homedir "$GPG_HOME" --allow-secret-key-import --import tests/files/secured

 wait

-exec python -m securedrop_client --sdc-home "$SDC_HOME" --no-proxy $@
+exec python -m securedrop_client --sdc-home "$SDC_HOME" --no-proxy $@ &
+
+wait
```

Reply:
making the client a bg process and waiting at the end did the trick! :100: